src/abstract_charm.py

```diff
@@ -109,10 +109,9 @@ def _exposed_read_write_endpoint(self) -> str:
     def _exposed_read_only_endpoint(self) -> str:
         """The exposed read-only endpoint"""
 
-    @property
     @abc.abstractmethod
-    def is_exposed(self) -> typing.Optional[bool]:
-        """Whether router is exposed externally"""
+    def is_externally_accessible(self, event=None) -> typing.Optional[bool]:
+        """Whether router is externally accessible"""
```

Comment:
yes, it is only used by the vm charm. but needs to be defined in abstract_charm because reconcile in workload calls to it. in vm, it will return `True/False`, in k8s it will return `None`

Reply:
done in 0978187f9b60c3c39d9f8f7cfbb39011ec0cdb90